conans/model/profile.py

```diff
@@ -12,15 +13,21 @@ class Profile(object):
 
     def __init__(self):
         # Sections
+        self.processed_settings = None
         self.settings = OrderedDict()
         self.package_settings = defaultdict(OrderedDict)
         self.env_values = EnvValues()
         self.options = OptionsValues()
         self.build_requires = OrderedDict()  # conan_ref Pattern: list of conan_ref
 
-    @property
-    def settings_values(self):
-        return Values.from_list(list(self.settings.items()))
+    def process_settings(self, client_cache, preprocess=True):
+        self.processed_settings = client_cache.settings.copy()
+        self.processed_settings.values = Values.from_list(list(self.settings.items()))
+        if preprocess:
```

Comment:
I don't understand what is the preprocess

Reply:
The preprocess is to automatically define MD/MDd for release/debug build_type.

Reply:
oh! right. that preprocess. ok